peak/adt.py

```diff
@@ -8,7 +8,7 @@
 import weakref
 import dataclasses as dc
 
-__all__ =  ['new', 'Product', 'is_product', 'Tuple', ]
+__all__ =  ['new', 'ISABuilder', 'Product', 'is_product', 'Tuple', ]
```

Comment:
Why do you need ISABuilder exported? It should not be extended by user code

Reply:
I need to be able to check if an object is an ISABuilder type and/or is a class that inherits from ISABuilder. Im fine with a helper function instead.